stx-rosetta.Dockerfile

```diff
@@ -1,147 +1,124 @@
-  && chmod +x stacks-blockchain-api/stacks_api
-ENV PATH="$PATH:$HOME/stacks-blockchain-api"
-EXPOSE 3999
-
-### Install Postgres
-RUN sudo apt-get install -y postgresql-12 postgresql-contrib-12
-
-### Setup Postgres
-# Borrowed from https://github.com/gitpod-io/workspace-images/blob/master/postgres/Dockerfile
-ENV PATH="$PATH:/usr/lib/postgresql/12/bin"
-ENV PGDATA="/data/stacky/.pgsql/data"
-RUN mkdir -p ~/.pg_ctl/bin ~/.pg_ctl/sockets \
-  && printf '#!/bin/bash\n[ ! -d $PGDATA ] && mkdir -p $PGDATA && initdb -D $PGDATA\npg_ctl -D $PGDATA -l ~/.pg_ctl/log -o "-k ~/.pg_ctl/sockets" start\n' > ~/.pg_ctl/bin/pg_start \
-  && printf '#!/bin/bash\npg_ctl -D $PGDATA -l ~/.pg_ctl/log -o "-k ~/.pg_ctl/sockets" stop\n' > ~/.pg_ctl/bin/pg_stop \
-  && chmod +x ~/.pg_ctl/bin/*
-ENV PATH="$PATH:$HOME/.pg_ctl/bin"
-
-### Clear caches
-RUN sudo apt-get clean && sudo rm -rf /var/cache/apt/* /var/lib/apt/lists/* /tmp/*
-
-### Setup service env vars
+ARG STACKS_LOG_DIR=/var/log/stacks-node
+ARG STACKS_SVC_DIR=/etc/service
+ARG STACKS_BLOCKCHAIN_DIR=/stacks-blockchain
+ARG STACKS_BLOCKCHAIN_API_DIR=/stacks-blockchain-api
+ARG PG_DATA=/data/postgres
+ARG PG_VERSION=12
 ENV PG_HOST=127.0.0.1
 ENV PG_PORT=5432
-ENV PG_USER=stacky
+ENV PG_USER=postgres
 ENV PG_PASSWORD=postgres
 ENV PG_DATABASE=postgres
-
+ENV PG_DATA=${PG_DATA}
+ENV PG_VERSION=${PG_VERSION}
+ENV STACKS_SVC_DIR=${STACKS_SVC_DIR}
+ENV STACKS_BLOCKCHAIN_DIR=${STACKS_BLOCKCHAIN_DIR}
+ENV STACKS_BLOCKCHAIN_API_DIR=${STACKS_BLOCKCHAIN_API_DIR}
+ENV STACKS_NETWORK=${STACKS_NETWORK}
+ENV STACKS_LOG_DIR=${STACKS_LOG_DIR}
 ENV STACKS_CORE_EVENT_PORT=3700
 ENV STACKS_CORE_EVENT_HOST=127.0.0.1
-ENV STACKS_NETWORK=$STACKS_NETWORK
-
 ENV STACKS_EVENT_OBSERVER=127.0.0.1:3700
-
 ENV STACKS_BLOCKCHAIN_API_PORT=3999
 ENV STACKS_BLOCKCHAIN_API_HOST=0.0.0.0
-
 ENV STACKS_CORE_RPC_HOST=127.0.0.1
 ENV STACKS_CORE_RPC_PORT=20443
-
-### Startup script & coordinator
-RUN printf '#!/bin/bash\n\
-trap "exit" INT TERM\n\
-trap "kill 0" EXIT\n\
-echo Your container args are: "$@"\n\
-tail --retry -F stacks-api.log stacks-node.log 2>&1 &\n\
-while true\n\
-do\n\
-  pg_start\n\
-  stacks_api &> stacks-api.log &\n\
-  stacks_api_pid=$!\n\
-  if [ $STACKS_NETWORK = "mocknet" -o $STACKS_NETWORK = "dev" ]; then\n\
-    stacks-node start --config=/data/stacky/Stacks-${STACKS_NETWORK}.toml &> stacks-node.log &\n\
-  elif [ $STACKS_NETWORK = "testnet"]; then \n\
-    stacks-node start --config=/data/stacky/Stacks-mocknet.toml &> stacks-node.log &\n\
-  else\n\
-    stacks-node mainnet &> stacks-node.log &\n\
-  fi\n\
-  stacks_node_pid=$!\n\
-  wait $stacks_node_pid\n\
-  echo "node exit, restarting..."\n\
-  rkill -9 $stacks_api_pid\n\
-  pg_stop\n\
-  rm -rf $PGDATA\n\
-  sleep 5\n\
-done\n\
-' >> run.sh && chmod +x run.sh
-
+ENV MAINNET_STACKS_CHAIN_ID=0x00000001
+ENV TESTNET_STACKS_CHAIN_ID=0x80000000
+ENV V2_POX_MIN_AMOUNT_USTX=90000000260
```

Comment:
Can we combine all of these `ENV` directives into one directive? It'll result in fewer layers in the image and decrease its size. Also if any of them aren't used in the final image's runtime, you can just use their `ARG` counterpart due to the same reason in my other comment.

Reply:
Yes, but it would also be incredibly hard to read. the goal here isn't a small image

Reply:
That's fine if you want to keep it that way, I don't find it difficult to read with one directive. Also it seems `ENV` directives don't create additional layers anymore either which is nice.